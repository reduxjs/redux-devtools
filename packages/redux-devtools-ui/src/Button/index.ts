export { default } from './Button.js';
