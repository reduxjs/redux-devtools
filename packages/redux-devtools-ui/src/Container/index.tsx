import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { useTheme, ThemeData } from '../utils/theme';
import { MainContainerWrapper, ContainerWrapper } from './styles';
import { Theme } from '../themes/default';

interface ContainerFromThemeDataProps {
  children?: React.ReactNode;
  themeData: ThemeData;
  className?: string;
}

const ContainerFromThemeData: React.FunctionComponent<
  ContainerFromThemeDataProps
> = ({ themeData, className, children }) => {
  const theme = useTheme(themeData);
  return (
    <ThemeProvider theme={theme}>
      <MainContainerWrapper className={className}>
        {children}
      </MainContainerWrapper>
    </ThemeProvider>
  );
};

interface Props {
  children?: React.ReactNode;
  themeData?: ThemeData;
  theme?: Theme;
  className?: string;
}

const Container: React.FunctionComponent<Props> = ({
  themeData,
  className,
  theme,
  children,
}) => {
  if (!themeData) {
    return (
      <ContainerWrapper className={className} theme={theme}>
        {children}
      </ContainerWrapper>
    );
  }

  return (
    <ContainerFromThemeData themeData={themeData} className={className}>
      {children}
    </ContainerFromThemeData>
  );
};

Container.propTypes = {
  children: PropTypes.node,
  themeData: PropTypes.any,
  theme: PropTypes.any,
  className: PropTypes.string,
};

export default Container;
