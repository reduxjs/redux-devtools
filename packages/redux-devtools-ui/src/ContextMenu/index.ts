export { default } from './ContextMenu.js';
