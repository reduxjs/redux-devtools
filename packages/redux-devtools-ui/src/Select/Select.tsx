import React, { PureComponent, Component, ReactElement } from 'react';
import PropTypes from 'prop-types';
import ReactSelect, {
  GroupBase,
  Props as ReactSelectProps,
} from 'react-select';
import createThemedComponent from '../utils/createThemedComponent';
import { Theme } from '../themes/default';

export interface SelectProps<
  Option,
  IsMulti extends boolean = false,
  Group extends GroupBase<Option> = GroupBase<Option>
> extends Omit<ReactSelectProps<Option, IsMulti, Group>, 'theme'> {
  theme: Theme;
}

/**
 * Wrapper around [React Select](https://github.com/JedWatson/react-select).
 */
export class Select<
  Option,
  IsMulti extends boolean = false,
  Group extends GroupBase<Option> = GroupBase<Option>
> extends (PureComponent || Component)<SelectProps<Option, IsMulti, Group>> {
  render() {
    return (
      <ReactSelect
        {...this.props}
        theme={(theme) => ({
          ...theme,
          borderRadius: 0,
          colors: {
            ...theme.colors,

            primary: this.props.theme.base05,
            primary50: this.props.theme.base03,
            primary25: this.props.theme.base01,

            dangerLight: this.props.theme.base03,
            danger: this.props.theme.base07,

            neutral0: this.props.theme.base00,
            neutral5: this.props.theme.base01,
            neutral10: this.props.theme.base02,
            neutral20: this.props.theme.base03,
            neutral30: this.props.theme.base04,
            neutral40: this.props.theme.base05,
            neutral60: this.props.theme.base06,
            neutral80: this.props.theme.base07,
          },
          spacing: {
            ...theme.spacing,
            baseUnit: 2,
            controlHeight: this.props.theme.inputHeight,
          },
        })}
        styles={{
          container: (base) => ({
            ...base,
            flexGrow: 1,
          }),
          control: (base, props) => ({
            ...base,
            backgroundColor: props.isFocused
              ? props.theme.colors.neutral10
              : props.theme.colors.neutral5,
            borderColor: props.theme.colors.neutral10,

            '&:hover': {
              backgroundColor: props.theme.colors.neutral10,
              borderColor: props.theme.colors.neutral10,
            },
          }),
          menu: (base) => ({
            ...base,
            zIndex: 10,
          }),
        }}
      />
    );
  }

  static propTypes = {
    isClearable: PropTypes.bool, // should it be possible to reset value
    isDisabled: PropTypes.bool, // whether the Select is disabled or not
    isLoading: PropTypes.bool, // whether the Select is loading externally or not
    maxMenuHeight: PropTypes.number, // maximum css height for the opened menu of options
    isMulti: PropTypes.bool, // multi-value input
    isSearchable: PropTypes.bool, // whether to enable searching feature or not
    value: PropTypes.any, // initial field value
    menuPlacement: PropTypes.oneOf(['auto', 'bottom', 'top']), // value to control the opening direction
  };
}

export interface ExternalSelectProps<
  Option,
  IsMulti extends boolean = false,
  Group extends GroupBase<Option> = GroupBase<Option>
> extends Omit<ReactSelectProps<Option, IsMulti, Group>, 'theme'> {
  theme?: Theme;
}

type SelectComponent = <
  Option,
  IsMulti extends boolean = false,
  Group extends GroupBase<Option> = GroupBase<Option>
>(
  props: ExternalSelectProps<Option, IsMulti, Group>
) => ReactElement;

export default createThemedComponent(Select) as SelectComponent & {
  theme?: Theme;
};
