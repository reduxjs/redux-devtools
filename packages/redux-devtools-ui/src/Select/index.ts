export { default } from './Select.js';
