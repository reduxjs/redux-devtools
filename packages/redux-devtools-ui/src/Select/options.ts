export const options = [
  { value: 'one', label: 'One' },
  { value: 'two', label: 'Two' },
  { value: 'hundred', label: 'One hundred' },
];
