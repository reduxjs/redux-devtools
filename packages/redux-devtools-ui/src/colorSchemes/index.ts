export { default } from './default';
export { default as github } from './github';
export { default as atomOneDark } from './atom-one-dark';
export { default as dracula } from './dracula';
export { default as iRBlack } from './ir-black';
export { default as macintosh } from './macintosh';
export { default as materia } from './materia';
export { default as oceanicNext } from './oceanic-next';
export { default as phD } from './phd';
export { default as pico } from './pico';
export { default as solarFlare } from './solar-flare';
export { default as spacemacs } from './spacemacs';
export { default as unikitty } from './unikitty';
export { default as woodland } from './woodland';
