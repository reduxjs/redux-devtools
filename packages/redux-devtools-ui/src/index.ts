export { default as Container } from './Container';
export { default as Button } from './Button';
export { default as ContextMenu } from './ContextMenu';
export { default as Dialog } from './Dialog';
export { default as Editor } from './Editor';
export { default as Form } from './Form';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Tabs, Tab } from './Tabs';
export { default as SegmentedControl } from './SegmentedControl';
export { default as Notification } from './Notification';
export * from './Toolbar';

import color from './utils/color';
export const effects = { color };
export { default as createStyledComponent } from './utils/createStyledComponent';
export { Theme, ThemeFromProvider, Scheme } from './utils/theme';
