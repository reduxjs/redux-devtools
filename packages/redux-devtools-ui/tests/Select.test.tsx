import React from 'react';
import { render, mount, CommonWrapper, ReactWrapper } from 'enzyme';
import { renderToJson, mountToJson } from 'enzyme-to-json';
import { Select } from '../src';
import { options } from '../src/Select/options';

describe('Select', function () {
  it('renders correctly', () => {
    const wrapper = render(
      <Select
        options={options}
        onChange={() => {
          // noop
        }}
      />
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders with props', () => {
    const wrapper = render(
      <Select
        options={options}
        onChange={() => {
          // noop
        }}
        value={options.filter((option) => option.value === 'one')}
        maxMenuHeight={20}
        isClearable
        isDisabled
        isLoading
        isMulti
        isSearchable={false}
        menuPlacement="top"
      />
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('should select another option', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Select options={options} onInputChange={onChange} />
    );

    const input = wrapper.find('input');
    (input.at(0).instance() as unknown as HTMLInputElement).value = 'two';
    input.first().simulate('change');
    expect(mountToJson(wrapper)).toMatchSnapshot();
    input.first().simulate('keyDown', { keyCode: 13 });
    expect(onChange).toBeCalled();
  });

  it("shouldn't find any results", () => {
    const onChange = jest.fn();
    const wrapper = mount(<Select options={options} onChange={onChange} />);

    const input = wrapper.find('input');
    (input.at(0).instance() as unknown as HTMLInputElement).value = 'text';
    input.first().simulate('change');
    expect(mountToJson(wrapper)).toMatchSnapshot(); // 'No results found'
    input.first().simulate('keyDown', { keyCode: 13 });
    expect(onChange).not.toBeCalled();
  });
});
