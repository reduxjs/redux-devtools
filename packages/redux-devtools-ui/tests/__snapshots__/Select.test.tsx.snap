// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  class=" css-18o3wl1-container"
>
  <span
    class="css-1f43avz-a11yText-A11yText"
    id="react-select-2-live-region"
  />
  <span
    aria-atomic="false"
    aria-live="polite"
    aria-relevant="additions text"
    class="css-1f43avz-a11yText-A11yText"
  />
  <div
    class=" css-gdsksj-control"
  >
    <div
      class=" css-1dzc0nd-ValueContainer"
    >
      <div
        class=" css-du77wi-placeholder"
        id="react-select-2-placeholder"
      >
        Select...
      </div>
      <div
        class=" css-1qx78vi-Input"
        data-value=""
      >
        <input
          aria-autocomplete="list"
          aria-controls="react-select-2-listbox"
          aria-describedby="react-select-2-placeholder"
          aria-expanded="false"
          aria-haspopup="true"
          aria-owns="react-select-2-listbox"
          autocapitalize="none"
          autocomplete="off"
          autocorrect="off"
          class=""
          id="react-select-2-input"
          role="combobox"
          spellcheck="false"
          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
          tabindex="0"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class=" css-1hb7zxy-IndicatorsContainer"
    >
      <span
        class=" css-1urf5ax-indicatorSeparator"
      />
      <div
        aria-hidden="true"
        class=" css-aazq4s-indicatorContainer"
      >
        <svg
          aria-hidden="true"
          class="css-tj5bde-Svg"
          focusable="false"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select renders with props 1`] = `
<div
  class=" css-lk03f7-container"
>
  <span
    class="css-1f43avz-a11yText-A11yText"
    id="react-select-3-live-region"
  />
  <span
    aria-atomic="false"
    aria-live="polite"
    aria-relevant="additions text"
    class="css-1f43avz-a11yText-A11yText"
  />
  <div
    class=" css-gdsksj-control"
  >
    <div
      class=" css-be27w2-ValueContainer"
    >
      <div
        class="css-168vvpw-multiValue"
      >
        <div
          class="css-1m51f4l"
        >
          One
        </div>
        <div
          aria-label="Remove One"
          class="css-1to589f"
          role="button"
        >
          <svg
            aria-hidden="true"
            class="css-tj5bde-Svg"
            focusable="false"
            height="14"
            viewBox="0 0 20 20"
            width="14"
          >
            <path
              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
            />
          </svg>
        </div>
      </div>
      <input
        aria-autocomplete="list"
        aria-controls="react-select-3-listbox"
        aria-expanded="false"
        aria-haspopup="true"
        aria-owns="react-select-3-listbox"
        aria-readonly="true"
        class="css-mohuvp-dummyInput-DummyInput"
        disabled=""
        id="react-select-3-input"
        inputmode="none"
        role="combobox"
        tabindex="0"
        value=""
      />
    </div>
    <div
      class=" css-1hb7zxy-IndicatorsContainer"
    >
      <div
        aria-hidden="true"
        class=" css-1gilzap-loadingIndicator"
      >
        <span
          class="css-1xtdfmb-LoadingDot"
        />
        <span
          class="css-zoievk-LoadingDot"
        />
        <span
          class="css-x748d8-LoadingDot"
        />
      </div>
      <span
        class=" css-1urf5ax-indicatorSeparator"
      />
      <div
        aria-hidden="true"
        class=" css-aazq4s-indicatorContainer"
      >
        <svg
          aria-hidden="true"
          class="css-tj5bde-Svg"
          focusable="false"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select should select another option 1`] = `
<div
  class=" css-18o3wl1-container"
>
  <span
    class="css-1f43avz-a11yText-A11yText"
    id="react-select-4-live-region"
  />
  <span
    aria-atomic="false"
    aria-live="polite"
    aria-relevant="additions text"
    class="css-1f43avz-a11yText-A11yText"
  >
    <span
      id="aria-selection"
    />
    <span
      id="aria-context"
    >
      option Two focused, 2 of 3. 1 result available for search term two. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.
    </span>
  </span>
  <div
    class=" css-1gv8qk2-control"
  >
    <div
      class=" css-1dzc0nd-ValueContainer"
    >
      <div
        class=" css-1qx78vi-Input"
        data-value="two"
      >
        <input
          aria-autocomplete="list"
          aria-controls="react-select-4-listbox"
          aria-describedby="react-select-4-placeholder"
          aria-expanded="true"
          aria-haspopup="true"
          aria-owns="react-select-4-listbox"
          autocapitalize="none"
          autocomplete="off"
          autocorrect="off"
          class=""
          id="react-select-4-input"
          role="combobox"
          spellcheck="false"
          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
          tabindex="0"
          type="text"
          value="two"
        />
      </div>
    </div>
    <div
      class=" css-1hb7zxy-IndicatorsContainer"
    >
      <span
        class=" css-1urf5ax-indicatorSeparator"
      />
      <div
        aria-hidden="true"
        class=" css-aazq4s-indicatorContainer"
      >
        <svg
          aria-hidden="true"
          class="css-tj5bde-Svg"
          focusable="false"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class=" css-17wpf85-menu"
    id="react-select-4-listbox"
  >
    <div
      class=" css-87spde-MenuList"
    >
      <div
        aria-disabled="false"
        class=" css-1npzjqc-option"
        id="react-select-4-option-1"
        tabindex="-1"
      >
        Two
      </div>
    </div>
  </div>
</div>
`;

exports[`Select shouldn't find any results 1`] = `
<div
  class=" css-18o3wl1-container"
>
  <span
    class="css-1f43avz-a11yText-A11yText"
    id="react-select-5-live-region"
  />
  <span
    aria-atomic="false"
    aria-live="polite"
    aria-relevant="additions text"
    class="css-1f43avz-a11yText-A11yText"
  >
    <span
      id="aria-selection"
    />
    <span
      id="aria-context"
    >
       0 results available for search term text. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.
    </span>
  </span>
  <div
    class=" css-1gv8qk2-control"
  >
    <div
      class=" css-1dzc0nd-ValueContainer"
    >
      <div
        class=" css-1qx78vi-Input"
        data-value="text"
      >
        <input
          aria-autocomplete="list"
          aria-controls="react-select-5-listbox"
          aria-describedby="react-select-5-placeholder"
          aria-expanded="true"
          aria-haspopup="true"
          aria-owns="react-select-5-listbox"
          autocapitalize="none"
          autocomplete="off"
          autocorrect="off"
          class=""
          id="react-select-5-input"
          role="combobox"
          spellcheck="false"
          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
          tabindex="0"
          type="text"
          value="text"
        />
      </div>
    </div>
    <div
      class=" css-1hb7zxy-IndicatorsContainer"
    >
      <span
        class=" css-1urf5ax-indicatorSeparator"
      />
      <div
        aria-hidden="true"
        class=" css-aazq4s-indicatorContainer"
      >
        <svg
          aria-hidden="true"
          class="css-tj5bde-Svg"
          focusable="false"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class=" css-17wpf85-menu"
    id="react-select-5-listbox"
  >
    <div
      class=" css-87spde-MenuList"
    >
      <div
        class=" css-36co7-NoOptionsMessage"
      >
        No options
      </div>
    </div>
  </div>
</div>
`;
